src/pkg/k8s/session/session.go

```diff
@@ -212,3 +218,80 @@ func (ks *K8sSession) ServeConnStatus() http.HandlerFunc {
 		}
 	}
 }
+
+func (ks *K8sSession) Switch() http.HandlerFunc {
+	return func(w http.ResponseWriter, r *http.Request) {
+		if ks.InCluster {
+			http.Error(w, "cannot switch clusters while in-cluster", http.StatusBadRequest)
+			return
+		}
+
+		body, err := io.ReadAll(r.Body)
+		if err != nil {
+			http.Error(w, "failed to read request body", http.StatusBadRequest)
+			return
+		}
+		defer r.Body.Close()
+
+		var reqData map[string]client.ClusterInfo
+		if err := json.Unmarshal(body, &reqData); err != nil {
+			http.Error(w, "invalid request format", http.StatusBadRequest)
+			return
+		}
```

Comment:
nit: I think we can simplify this like so: 

```suggestion
		defer r.Body.Close()

		var reqData map[string]client.ClusterInfo
		if err := json.NewDecoder(r.Body).Decode(&reqData); err != nil {
			http.Error(w, "invalid request format", http.StatusBadRequest)
			return
		}
```